src/mscorlib/corefx/System/Buffers/TlsOverPerCoreLockedStacksArrayPool.Unix.cs

```diff
@@ -0,0 +1,28 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Win32;
+using System.Runtime.CompilerServices;
+
+namespace System.Buffers
+{
+    internal sealed partial class TlsOverPerCoreLockedStacksArrayPool<T>
+    {
+        /// <summary>Get an identifier for the current thread to use to index into the stacks.</summary>
+        private static int ExecutionId
+        {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            get
+            {
+                // On Unix, GetCurrentProcessorNumber is implemented in terms of sched_getcpu, which
```

Comment:
I wonder what the performance characteristics of GetCurrentProcessorNumber are accross different Unix OSes where it is actually implemented. We may want to cache it - similar to how GC does it; or maybe piggy back on the GC cache where possible.